### PR TITLE
Implementaion of Settings

### PR DESCRIPTION
**Pull request by baquer**
Implementation Of Setting Area.

### CODE REVIEW

File: AmahiAnywhere/AmahiAnywhere/MenuTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+//
+
+import UIKit
+
+class MenuTableViewCell: UITableViewCell {
+
+    @IBOutlet weak var lblMenuName: UILabel!
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
```
Comment (codedentwickler) [inline AmahiAnywhere/AmahiAnywhere/MenuTableViewCell.swift]: You can remove the generated methods if they are not used

File: AmahiAnywhere/AmahiAnywhere/MenuViewController.swift [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+//  Created by Syed on 24/03/18.
+//  Copyright © 2018 Amahi. All rights reserved.
+//
+
+import UIKit
+import MessageUI
+
+class MenuViewController: UIViewController,UITableViewDelegate,UITableViewDataSource,MFMailComposeViewControllerDelegate {
+
+    var ManuNameArray:Array = [String]()
```
Comment (codedentwickler) [inline AmahiAnywhere/AmahiAnywhere/MenuViewController.swift]: Kindly rename ManuNameArray to something like menuActionsTitle.

File: AmahiAnywhere/AmahiAnywhere/MenuViewController.swift [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+                refreshAlert .dismiss(animated: true, completion: nil)
+                
+                
+            }))
+            
+            present(refreshAlert, animated: true, completion: nil)
+        }
+        if cell.lblMenuName.text! == "Connection"
+        {
+            performSegue(withIdentifier: "connection", sender: nil)
```
Comment (codedentwickler) [inline AmahiAnywhere/AmahiAnywhere/MenuViewController.swift]: Add this identifier to SegueIdentifiers.

File: AmahiAnywhere/AmahiAnywhere/MenuViewController.swift [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    // MARK:- Delegates Method
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return ManuNameArray.count
+        
+    }
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "MenuTableViewCell", for: indexPath) as! MenuTableViewCell
+        
+
```
Comment (codedentwickler) [inline AmahiAnywhere/AmahiAnywhere/MenuViewController.swift]: Too much space

File: AmahiAnywhere/AmahiAnywhere/MenuViewController.swift [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+            self.present(myalert, animated: true)
+            
+            
+        }
+        if cell.lblMenuName.text! == "Rate"
+        {
+            
+            
+            let appID = StringLiterals.APP_ID
+            let urlStr = "itms-apps://itunes.apple.com/app/viewContentsUserReviews?id=\(appID)"
```
Comment (codedentwickler) [inline AmahiAnywhere/AmahiAnywhere/MenuViewController.swift]: You can move this URL to StringLiterals and mark it with something like `User Review URL`

File: AmahiAnywhere/AmahiAnywhere/MenuViewController.swift [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    }
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        
+        tableView.deselectRow(at: indexPath, animated: true)
+        let revealviewcontroller:SWRevealViewController = self.revealViewController()
+        
+        let cell:MenuTableViewCell = tableView.cellForRow(at: indexPath) as! MenuTableViewCell
+        
+        if cell.lblMenuName.text! == "About"
+        {
```
Comment (codedentwickler) [inline AmahiAnywhere/AmahiAnywhere/MenuViewController.swift]: I think we can do replace this `if`s with a switch. I personally won't advice doing a strict menu name check.

File: AmahiAnywhere/AmahiAnywhere/MenuTableViewCell.swift [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+//
+//  Created by Syed on 24/03/18.
+//  Copyright © 2018 Amahi. All rights reserved.
+//
+
+import UIKit
+
+class MenuTableViewCell: UITableViewCell {
+
+    @IBOutlet weak var lblMenuName: UILabel!
```
Comment (codedentwickler) [inline AmahiAnywhere/AmahiAnywhere/MenuTableViewCell.swift]: No need to add Menu in the variable name this is already a menu cell. titleLabel is more appropriate. Generally, avoid using abbreviations, you should use a full description of the UI widget as a subscript of your variable names. Check out naming conventions in the swift styles shared earlier https://github.com/raywenderlich/swift-style-guide

**Comment by codedentwickler** [comment]
@baquer Well done on this PR. I noticed the settings area is in a Navigation Drawer(Kindly correct me if I'm wrong). This is strange as I have not seen this implemented like this on any app. We are not navigating anywhere in the settings area. I think a Simple gear button in the top right()someone did this earlier or a Settings entry in the Home Navigation Drawer if we have one eventually should launch the settings area.
The Settings area can just be a simple UITableView with the actions you implemented. 
Currently, we have a Navigation controller on Dashboard, pushing the Settings View Controller instead of presenting it will automatically show a `Back` button to get back to the Dashboard after the user is done in the Settings area. 
You can check out the default Settings App on iOS to see it is implemented.

**Comment by csoni111** [comment]
Also I was unable to build the app. I think the header files, `SWRevealViewController.h` and `SWRevealViewController.m` are missing.

**Comment by baquer** [comment]
@codedentwickler  Yes you are right , we have sign out button in dashboardVC so can I replace the sign out button with setting button ? Also when we connect the settingVc with dashboard it will not automatically show a Back button to get back to the Dashboard , because the menu button is in the left hand side , so we have only two options, either we move the menu to right side or make a button that directs to home/dashboard as I done  earlier.

**Comment by baquer** [comment]
@csoni111  i am checking the missing files .

**Comment by codedentwickler** [comment]
@baquer You should push the settings View controller instead of presenting it. Pushing it will automatically show the back button. You can rename the back button text in the settings View Controller to some like `Home` or simply `Back`.

**Comment by codedentwickler** [comment]
@baquer I still can't build. The files are missing. It will work on your end because you are referencing the files from a direct path on your PC. You need to move the files to the project and use a relative path to it and add it to git. Xcode is trying to find this path `.../SWRevealViewController-master/SWRevealViewController/SWRevealViewController.h` on my PC. The folder only exists on your PC.

**Comment by baquer** [comment]
while building the app this gives an error Error “_OBJC_CLASS_$_AppDelegate”, referenced from:“ and error: linker command failed with exit code 1 (use -v to see invocation)”

You will need to make sure that your SWRevealViewController.m and SWRevealViewController.h are added under Compile Sources in your target settings.

To do this, click on the project (left upper on your Xcode window) -> Click on Build Phases tab -> Compile Sources and then check to see whether AppDelegate.m is actually added to the list. If not, click + button -> Type SWRevealViewController.m and SWRevealViewController.h -> Select and then Add it. That should do.

**Comment by codedentwickler** [comment]
I noticed you just pushed the files in your last commit. I will check this out. I (everyone) should not have to change anything in my build settings to make this work. I will check it out again.

**Comment by baquer** [comment]
Yes actually when I directly pushed the file , it's get deleted automatically while pushing , so that's why I again pushed the file in the last commit by putting the files in a folder.

**Comment by codedentwickler** [comment]
Hi, @baquer I noticed you added SWRevealViewController because of the side menu, it is a lot of code and unnecessary. I don't think we need a side menu for settings. A simple UITableView like what we have on the Android app will do. It will have a back button to get back to Home. If we should have a Side Menu, It should be on the Dashboard. This is a screenshot of the Android Settings UI. You can use it as a guide. 
screenshot_2018-03-26-17-24-05

**Comment by baquer** [comment]
yes @codedentwickler  i used SWRevealViewController for the side menu , because in my previous PR @cpg advised me to use a HamBurger type setting menu. So if team members wants a simple tableView for the setting area, i can close this PR and make a new PR thats contains a tableview for the setting area .
 Hoping for a favourable reply from the members.

**Comment by codedentwickler** [comment]
I think you misinterpreted @cpg comment here https://github.com/amahi/ios/issues/6. He gave feedback on the colour, and that no menu should be added. His specification actually follows the one implemented on Android. He 

> There is browsing functionality on the app, so it 1) should not be broken by this 2) it should be integrated smoothly so the user knows what to do to get back to the rest of the app from here.

Pushing the Setting View controller and Navigating back with the Back button will return the User to the previous flow. I think implementing it the way it works on Android will fulfil @cpg wishes above.

**Comment by baquer** [comment]
okay I get the points , so let me resolve this issue .
@codedentwickler  can i close this PR and make a new according to #6 requirements ?

**Pull request closed without merging**